src/run.ts

```diff
@@ -47,10 +54,155 @@ const createRelease = async (
+           get the issue number.
+        6. Create a comment for each issue and PR
+      */
+
+  const repo = github.context.repo;
+
+  let tagPage = 0;
+  let tagFound = false;
+  let tagCommitSha = "";
+
+  /* 1 */
+  while (!tagFound) {
+    await octokit.repos
+      .listTags({
+        ...repo,
+        per_page: 100,
+        page: tagPage,
+      })
+      .then(({ data }) => {
+        const tag = data.find((el) => el.name === tagName);
+        if (tag) {
+          tagFound = true;
+          /* 2 */
+          tagCommitSha = tag.commit.sha;
+        }
+        tagPage += 1;
+      })
+      .catch((err) => console.warn(err));
+  }
+
+  /* 3 */
+  const commits = await octokit.repos
+    .listCommits({
+      ...repo,
+      sha: tagCommitSha,
+    })
+    .then(({ data }) => data);
+
+  const shas = commits.map(({ sha }) => sha);
+
+  /* Build a seach query to retrieve pulls with commit hashes.
+   *  example: repo:<OWNER>/<REPO>+type:pr+is:merged+hash:<FIRST_COMMIT_HASH>+hash:<SECOND_COMMIT_HASH>...
+   */
+  const searchQueries = getSearchQueries(
+    `repo:${repo.owner}/${repo.repo}+type:pr+is:merged`,
+    shas
+  ).map(
+    async (q) => (await octokit.search.issuesAndPullRequests({ q })).data.items
+  );
+
+  const queries = await (await Promise.all(searchQueries)).flat();
+
+  const queriesSet = queries.map((el) => el.number);
+
+  const filteredQueries = queries.filter(
+    (el, i) => queriesSet.indexOf(el.number) === i
+  );
+
+  /* 4 */
+  const pulls = await filteredQueries.filter(
+    async ({ number }) =>
+      (
+        await octokit.pulls.listCommits({
+          owner: repo.owner,
+          repo: repo.repo,
+          pull_number: number,
+        })
+      ).data.find(({ sha }) => shas.includes(sha)) ||
+      shas.includes(
+        (
+          await octokit.pulls.get({
+            owner: repo.owner,
+            repo: repo.repo,
+            pull_number: number,
+          })
+        ).data.merge_commit_sha
+      )
+  );
+
+  const parser = issueParser("github");
+
+  /* 5 */
+  const issues = [
+    ...pulls.map((pr) => pr.body),
+    ...commits.map(({ commit }) => commit.message),
+  ].reduce((issues, message) => {
+    return message
+      ? issues.concat(
+          parser(message)
+            .actions.close.filter(
+              (action) =>
+                action.slug === null ||
+                action.slug === undefined ||
+                action.slug === `${repo.owner}/${repo.repo}`
+            )
+            .map((action) => ({ number: Number.parseInt(action.issue, 10) }))
+        )
+      : issues;
+  }, [] as { number: number }[]);
+
+  /* 6 */
+  await Promise.all(
+    [...new Set([...pulls, ...issues].map(({ number }) => number))].map(
+      async (number) => {
+        const issueComment = {
+          ...repo,
+          issue_number: number,
+          body: getReleaseMessage(htmlUrl, tagName),
+        };
+
+        octokit.issues.createComment(issueComment);
```

Comment:
```suggestion
        return octokit.issues.createComment(issueComment);
```

Otherwise the outer `Promise.all` won't wait on those calls.